### PR TITLE
add survey to site

### PR DESCRIPTION
**Pull request by keertip**
@alan-knight 

Is there anyway to test if this does work?

### CODE REVIEW

**Comment by alan-knight** [comment]
What I do is make a new "version" on app engine of the site and upload to that and try it. It's painful, but works.

File: client/lib/transformer.dart [hunk 1]
```diff
@@ -154,3 +162,5 @@ const _ANALYTICS_CODE = '''<script>
   ga('send', 'pageview');
 
 </script>''';
+
+const _SURVEY_CODE = '<script async="" defer="" src="//survey.g.doubleclick.net/async_survey?site=4vq7qmqpfn4zjduve64c4h6lqa"></script>';
```
Comment (alan-knight) [inline client/lib/transformer.dart]: Is this encoding our site directly into the code? What happens on dartdocs.org? What happens on Angular docs, or some other site that's using this?

Comment (sethladd) [inline client/lib/transformer.dart]: only api.dartlang.org should get this snippet. I don't think we want this actually in dartdoc-viewer. Probably more of a build-process thing when we build the SDK.

Comment (alan-knight) [inline client/lib/transformer.dart]: So we should probably be doing the same thing as with the analytics - it looks for a file with the information that isn't in the dartdoc-viewer, but is present in the deployment setup.

Comment (keertip) [inline client/lib/transformer.dart]: So do we need to add a check elsewhere?

Comment (alan-knight) [inline client/lib/transformer.dart]: So what's happening with the analytics is that the transformer looks for a
file at _CONFIG_PATH, which is lib/config/config.yaml. That's not present
in the checking, there's a sample version of it, and it's in the .gitignore
so it doesn't get checked in by accident. It's there in my copy that I use
to actually update the site. (It's a very robust deployment process)

If it's there, then that sets the config variable. The verifier and
analytics plugin check that for their values, and only do something if it's
present.

So we could use the same file for survey information, or add a different
one, put our specific survey number in there, and only replace the code if
it's present.

**Comment by keertip** [comment]
@alan-knight  PTAL - made the suggested changes.

**Comment by sethladd** [comment]
Thanks to you both!

When would this land on api.dartlang.org ?  (no specific rush, but I want to time the survey)

**Comment by alan-knight** [comment]
I merged it and redeployed to the survey site. I can make that the live
instance any time you want.

**Comment by sethladd** [comment]
Any time should be fine. We have one survey in the queue. Thanks Alan!

**Pull request merged**